agent/handlers/agentapi/taskprotection/v1/handlers/handlers_test.go

```diff
@@ -19,26 +19,135 @@ import (
 	"io"
 	"net/http"
 	"net/http/httptest"
+	"reflect"
 	"testing"
+	"time"
 
+	"github.com/aws/amazon-ecs-agent/agent/api"
+	mock_api "github.com/aws/amazon-ecs-agent/agent/api/mocks"
 	"github.com/aws/amazon-ecs-agent/agent/api/task"
+	"github.com/aws/amazon-ecs-agent/agent/credentials"
+	mock_credentials "github.com/aws/amazon-ecs-agent/agent/credentials/mocks"
+	"github.com/aws/amazon-ecs-agent/agent/ecs_client/model/ecs"
 	"github.com/aws/amazon-ecs-agent/agent/engine/dockerstate"
 	mock_dockerstate "github.com/aws/amazon-ecs-agent/agent/engine/dockerstate/mocks"
 	v3 "github.com/aws/amazon-ecs-agent/agent/handlers/v3"
 	"github.com/aws/amazon-ecs-agent/agent/utils"
+	"github.com/aws/aws-sdk-go/aws"
 	"github.com/golang/mock/gomock"
 	"github.com/gorilla/mux"
+	"github.com/pkg/errors"
 	"github.com/stretchr/testify/assert"
 )
 
-// Tests the path for PutTaskProtection API
+const (
+	testAccessKey              = "accessKey"
+	testSecretKey              = "secretKey"
+	testSessionToken           = "sessionToken"
+	testCluster                = "cluster"
+	testRegion                 = "region"
+	testEndpoint               = "endpoint"
+	testTaskCredentialsId      = "taskCredentialsId"
+	testV3EndpointId           = "endpointId"
+	testTaskArn                = "taskArn"
+	testServiceName            = "serviceName"
+	testAcceptInsecureCert     = false
+	protectionEnabledFieldName = "ProtectionEnabled"
+	expiresInMinutesFieldName  = "ExpiresInMinutes"
+	testExpiresInMinutes       = 5
+	testProtectionEnabled      = true
+)
+
+// MockTaskProtectionClientFactory is a mock of TaskProtectionClientFactory interface
+type MockTaskProtectionClientFactory struct {
+	ctrl     *gomock.Controller
+	recorder *MockTaskProtectionClientFactoryMockRecorder
+}
+
+// MockECSClientMockRecorder is the mock recorder for MockECSClient
+type MockTaskProtectionClientFactoryMockRecorder struct {
+	mock *MockTaskProtectionClientFactory
+}
+
+func NewMockTaskProtectionClientFactory(ctrl *gomock.Controller) *MockTaskProtectionClientFactory {
+	mock := &MockTaskProtectionClientFactory{ctrl: ctrl}
+	mock.recorder = &MockTaskProtectionClientFactoryMockRecorder{mock}
+	return mock
+}
+
+func (m *MockTaskProtectionClientFactory) EXPECT() *MockTaskProtectionClientFactoryMockRecorder {
+	return m.recorder
+}
+
+// GetTaskCredentials mocks base method
+func (m *MockTaskProtectionClientFactory) getECSClient(arg0 credentials.Manager, arg1 *task.Task) (api.ECSTaskProtectionSDK, int, error) {
```

Comment:
This is hand written code. Autogenerate code are generated with https://github.com/aws/amazon-ecs-agent/blob/master/agent/api/generate_mocks.go and can be configured.

I didn't bother to generate it since it is only used here and not even exported, but let me know if you think otherwise and I should generate it.

Reply:
I see. We can have it in the same `handlers` package yet in a new file. Maybe using the autogeneration tool will help us in the future if there are any changes? We'll just run the generation command again and replace the file.

Reply:
Yeah, I can do that.